org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/DefaultGradleBuild.java

```diff
@@ -46,7 +49,17 @@ public DefaultGradleBuild(BuildConfiguration buildConfig) {
     @Override
     public void synchronize(NewProjectHandler newProjectHandler, CancellationTokenSource tokenSource, IProgressMonitor monitor) throws CoreException {
         SynchronizeGradleBuildsOperation syncOperation = SynchronizeGradleBuildsOperation.forSingleGradleBuild(this, newProjectHandler);
-        syncOperation.run(tokenSource, monitor);
+        try {
+            syncOperation.run(tokenSource, monitor);
+            GradleMarkerManager.clear(this);
+        } catch (Exception e) {
+            GradleMarkerManager.clear(this);
+            ToolingApiStatus status = ToolingApiStatus.from("Project synchronization" , e);
+            if (status.severityMatches(IStatus.WARNING | IStatus.ERROR)) {
+                GradleMarkerManager.addError(this, status);
+            }
+            throw e;
```

Comment:
⭕️ I don't think synchronization should throw an exception if the problem could be handled by adding an error marker. There might be some conditions where we really can't continue (like we can't even import the root project), but in most cases the error is not really in the synchronization process itself, but in the users's build.

Reply:
I added the rethrow so that the error dialog is still shown to the user. I planned to follow this order:
1. introduce error markers
2. allow importing root projects of broken builds
3. remove error dialog
The rethrow would be removed in step #3

Reply:
Sounds good to me.